tracer/test/Datadog.Trace.Tests/Logging/DatadogLoggingTests.cs

```diff
@@ -205,6 +208,38 @@ public void ErrorsDuringLogging_DontBubbleUp()
             mockLogger.Verify();
         }
 
+        [Fact]
+        public void DuringStartup_OldLogFilesGetDeleted()
+        {
+            var tempLogsDir = Path.Combine(Path.GetTempPath(), "Datadog .NET Tracer\\logs");
+            Directory.CreateDirectory(tempLogsDir);
+
+            // Creating log files that match expected formats
+            var logPath = Path.Combine(tempLogsDir, "DD-DotNet-Profiler-Native-1.log");
+            File.Create(logPath).Dispose();
+            File.SetLastWriteTime(logPath, DateTime.Now.AddDays(-36));
+
+            File.Create(Path.Combine(tempLogsDir, "dotnet-tracer-managed-2.log")).Dispose();
+            File.Create(Path.Combine(tempLogsDir, "dotnet-tracer-native-3.log")).Dispose();
+
+            for (int i = 0; i < 3; i++)
+            {
+                // Adding random files that don't match the pattern
+                var mockFilePath = Path.Combine(tempLogsDir, $"random-file-{i}.txt");
+                File.Create(mockFilePath).Dispose();
+                File.SetLastWriteTime(mockFilePath, DateTime.Now.AddDays(-31 * i));
+            }
+
+            DatadogLogging.CleanLogFiles(tempLogsDir);
+
+            var allFiles = Directory.EnumerateFiles(tempLogsDir).Count();
+            var nonTraceFiles = Directory.EnumerateFiles(tempLogsDir, "random-file-*").Count();
+
+            Directory.Delete(tempLogsDir, true);
+
+            Assert.True(allFiles - nonTraceFiles == 2);
```

Comment:
TBH I had a hard time understanding what the test assertion was trying to do. I got it eventually, but I think with some refactoring we could make it clearer and faster for readers to understand. Here are a couple of suggestions that came to mind, let me know what you think:

1. You're calculating the number of files after calling `CleanLogFiles`. Maybe we could also calculate it beforehand as well, so we could write some assertion like `deletedFileCount.Should().Be(1);`
2. You're calculating the `nonTraceFiles` by enumerating the files with a file pattern. Another way to do the assertions could be to do this for each of the file groups (DD-DotNet-Profiler-* and dotnet-tracer-*) and then assert the count for every file group (we're only expecting DD-DotNet-Profiler to have a count of 0).

Reply:
Makes sense thank you, I updated the code to match the second approach